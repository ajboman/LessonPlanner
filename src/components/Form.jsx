import { useState } from 'react';
import { Configuration, OpenAIApi } from 'openai';

const openAiKey = import.meta.env.OPEN_AI_LESSON_PLAN_KEY;

const Form = () => {
  const [formData, setFormData] = useState({
    grade: '',
    subject: '',
    duration: '',
    classSize: '',
    learningObjectives: '',
    resourcesAvailable: '',
    teachingStyle: '',
    assessmentMethods: '',
    priorKnowledge: '',
    studentNeeds: '',
    standards: '',
  });

  const [prompt, setPrompt] = useState("");
  const [apiResponse, setApiResponse] = useState("");
  const [loading, setLoading] = useState(false);

  const handleChange = (event) => {
    setFormData({
      ...formData,
      [event.target.name]: event.target.value,
    });
  };

  const createString = (data) => {
    let new_prompt = "Create a lesson plan that abides by the following requirements: "

    for (const item in data) {
      if (data[item]) {
        new_prompt += item + ": " + data[item] + ", ";
      }
    }

    if (new_prompt.endsWith(', ')) {
      new_prompt = new_prompt.slice(0, -2);
    }
    setPrompt(new_prompt);
  };


  const handleSubmit = async (event) => {
    event.preventDefault();
    createString(formData);
    setLoading(true);

    const APIBody = {
      "model": "text-davinci-003",
      "prompt": "Say this is a test",
      "max_tokens": 7,
      "temperature": 0,
      "top_p": 1.0,
      "frequency_penalty": 0.0,
      "presence_penalty": 0.0
    }

    await fetch("https://api.openai.com/v1/chat/completions", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "Authorization": "Bearer" + openAiKey,
      },
      body: JSON.stringify(APIBody)
    }).then((data) => {
      return data.json();
    }).then((data) => {
      console.log(data);
    });

    setLoading(false);
  };


  return (
    <section className='mt-16 w-full'>
      <form className='relative flex flex-col justify-center gap-4'
        onSubmit={handleSubmit}>
        <label>
          Grade:
          <input
            type='text'
            name='grade'
            placeholder='9th Grade'
            value={formData.grade}
            onChange={handleChange}
          />
        </label>
        <label>
          Subject:
          <input
            type='text'
            name='subject'
            placeholder='Language Arts'
            value={formData.subject}
            onChange={handleChange}
          />
        </label>
        <label>
          Duration:
          <input
            type='text'
            name='duration'
            placeholder='50 minutes'
            value={formData.duration}
            onChange={handleChange}
          />
        </label>
        <label>
          Class Size:
          <input
            type='text'
            name='classSize'
            placeholder='30'
            value={formData.classSize}
            onChange={handleChange}
          />
        </label>
        <label>
          Learning Objectives:
          <textarea
            name='learningObjectives'
            placeholder='Students should be able to identify and explain the use of at least three literary devices'
            value={formData.learningObjectives}
            onChange={handleChange}
          />
        </label>
        <label>
          Resources Available:
          <textarea
            name='resourcesAvailable'
            placeholder='"To Kill a Mockingbird" novels, digital whiteboard for presenting a PowerPoint on literary devices, computers with internet access for research.'
            value={formData.resourcesAvailable}
            onChange={handleChange}
          />
        </label>
        <label>
          Teaching Style:
          <textarea
            name='teachingStyle'
            placeholder='Lecture, Group discussions, Individual work'
            value={formData.teachingStyle}
            onChange={handleChange}
          />
        </label>
        <label>
          Assessment Methods:
          <textarea
            name='assessmentMethods'
            placeholder='Completion of the literary device worksheet'
            value={formData.assessmentMethods}
            onChange={handleChange}
          />
        </label>
        <label>
          Prior Knowledge:
          <textarea
            name='priorKnowledge'
            placeholder='Students have learned about various literary devices'
            value={formData.priorKnowledge}
            onChange={handleChange}
          />
        </label>
        <label>
          Student Needs:
          <textarea
            name='studentNeeds'
            placeholder='For students who may struggle with the pace, provide a reference sheet of literary devices. For English Language Learners or students with special needs, provide extra support as needed, such as a translated version of the text or additional time to complete the worksheet. For advanced students, include more complex literary devices in their assigned passages.'
            value={formData.studentNeeds}
            onChange={handleChange}
          />
        </label>
        <label>
          Standards:
          <textarea
            name='standards'
            placeholder='This lesson aligns with Common Core Standards for 9th Grade English'
            value={formData.standards}
            onChange={handleChange}
          />
        </label>
        <button type="submit" className='mb-10'>Submit</button>
      </form>
    </section>
  );
};

export default Form;
