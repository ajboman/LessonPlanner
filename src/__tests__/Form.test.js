import createOpenAICompletion from '../services/route';
import { render, fireEvent, waitFor } from '@testing-library/react';
import Form from '../components/Form';

jest.mock('../services/route', () => ({
  __esModule: true,
  default: jest.fn().mockResolvedValue({ text: 'test response' }),
}));

const mockSaveLesson = jest.fn();
const mockDeleteLesson = jest.fn();

describe('Form component', () => {
  let component;

  beforeEach(() => {
    component = render(<Form saveLesson={mockSaveLesson} deleteLesson={mockDeleteLesson} />);
  });

  test('renders correctly', () => {
    expect(component.container).toMatchSnapshot();
  });

  test('updates field value on change', async () => {
    const gradeInput = component.getByPlaceholderText('9th Grade');
    fireEvent.change(gradeInput, { target: { value: '10th Grade' } });
    expect(gradeInput.value).toBe('10th Grade');
  });

  test('form submission receives test response', async () => {
    // fill in the form
    fireEvent.change(component.getByPlaceholderText('9th Grade'), { target: { value: '10th Grade' } });
    // add similar fireEvent.change calls for other form inputs as needed
  
    const submitButton = component.getByText('Submit');
    fireEvent.click(submitButton);
  
    await waitFor(() => {
      const popupContent = component.getByText('test response');
      expect(popupContent).toBeInTheDocument();
    });
  });
  

  // add more tests here as needed
});
