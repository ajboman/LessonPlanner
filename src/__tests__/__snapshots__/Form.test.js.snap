// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders correctly 1`] = `
<div>
  <section
    class="mt-8 w-full flex justify-center"
  >
    <form
      class="grid grid-cols-1 gap-4 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 p-5"
    >
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        Grade
        :
        <div
          class="flex"
        >
          <div
            class="relative w-full"
          >
            <input
              class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 rounded-lg p-2.5 text-sm"
              name="grade"
              placeholder="9th Grade"
              type="text"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        Subject
        :
        <div
          class="flex"
        >
          <div
            class="relative w-full"
          >
            <input
              class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 rounded-lg p-2.5 text-sm"
              name="subject"
              placeholder="Language Arts"
              type="text"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        Duration
        :
        <div
          class="flex"
        >
          <div
            class="relative w-full"
          >
            <input
              class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 rounded-lg p-2.5 text-sm"
              name="duration"
              placeholder="50 minutes"
              type="text"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        ClassSize
        :
        <div
          class="flex"
        >
          <div
            class="relative w-full"
          >
            <input
              class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 rounded-lg p-2.5 text-sm"
              name="classSize"
              placeholder="30"
              type="text"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        LearningObjectives
        :
        <textarea
          class="block w-full rounded-lg border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 textarea"
          name="learningObjectives"
          placeholder="Students should be able to identify and explain the use of at least three literary devices"
        />
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        ResourcesAvailable
        :
        <textarea
          class="block w-full rounded-lg border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 textarea"
          name="resourcesAvailable"
          placeholder=""To Kill a Mockingbird" novels, digital whiteboard for presenting a PowerPoint on literary devices, computers with internet access for research."
        />
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        TeachingStyle
        :
        <textarea
          class="block w-full rounded-lg border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 textarea"
          name="teachingStyle"
          placeholder="Lecture, Group discussions, Individual work"
        />
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        AssessmentMethods
        :
        <textarea
          class="block w-full rounded-lg border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 textarea"
          name="assessmentMethods"
          placeholder="Completion of the literary device worksheet"
        />
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        PriorKnowledge
        :
        <textarea
          class="block w-full rounded-lg border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 textarea"
          name="priorKnowledge"
          placeholder="Students have learned about various literary devices"
        />
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        StudentNeeds
        :
        <textarea
          class="block w-full rounded-lg border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 textarea"
          name="studentNeeds"
          placeholder="For students who may struggle with the pace, provide a reference sheet of literary devices. For English Language Learners or students with special needs, provide extra support as needed, such as a translated version of the text or additional time to complete the worksheet. For advanced students, include more complex literary devices in their assigned passages."
        />
      </label>
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300"
        data-testid="flowbite-label"
      >
        ExtraOptions
        :
        <textarea
          class="block w-full rounded-lg border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 textarea"
          name="extraOptions"
          placeholder="If something does not fit in the above categories put it here."
        />
      </label>
      <button
        class="text-white bg-cyan-700 border border-transparent hover:bg-cyan-800 focus:ring-4 focus:ring-cyan-300 disabled:hover:bg-cyan-700 dark:bg-cyan-600 dark:hover:bg-cyan-700 dark:focus:ring-cyan-800 dark:disabled:hover:bg-cyan-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg mb-10 bg-black col-span-full hover:bg-gray-600"
        type="submit"
      >
        <span
          class="flex items-center rounded-md text-sm px-4 py-2"
        >
          Submit
        </span>
      </button>
    </form>
  </section>
</div>
`;
